rest/dbHandler_test.go

```diff
@@ -54,5 +42,94 @@ func TestDbHandler(t *testing.T) {
 			t.Error("Expected error")
 		}
 	})
+}
+
+// E2E test of all functions in DbHandler executed in following order:
+// InsertTrack -> GetTracks -> GetTrack -> DeleteTrack
+func TestIntegratedDbHandlerTest(t *testing.T) {
+	app, token := testutils.RetrieveFirebaseIDToken()
+	values := map[string]io.Reader{
+		"file": mustOpen("../testdata/testIgc.igc"),
+	}
+
+	r := CompleteRouterSetup(app)
+
+	// Set up insertTrack
+	req, err := createMultipart(values, "/insertTrack", "POST")
+	if err != nil {
+		t.Error("Could not create multipart")
+	}
+	req.Header.Set("token", token)
+
+	// Run insertTrack
+	ret := testutils.TestRoute(req, r, "InsertTrack", t, http.StatusOK)
+	var insertBody models.IgcMetadata
+	err = json.Unmarshal(ret, &insertBody)
+	if err != nil {
+		t.Error("Failed extracting metadata response of InsertTrack")
+	}
+	// insertTrack DONE
 
+	// Set up getTracks
+	req = httptest.NewRequest("GET", "/getTracks", nil)
+	req.Header.Set("token", token)
+	req.Header.Set("queryType", "Private")
+	req.Header.Set("orderDirection", "Desc")
+
+	// Run getTracks
```

Comment:
If you are to do multiple steps in a test which relies on each other, consider using t.Run("", func()) instead, as you can skip the rest if something fails, as well as separating the logic.

Reply:
While noted, in the event that getTracks or getTrack fail, that would make deleteTrack not run, not cleaning up the database from the insertTrack. In addition, each step provides variables that the next test depends on, which is also the point of the test, confirming that everything works as expected. Tried creating 2 t.Run functions, making a variable in one and trying to make use of it in another, and I get errors on trying to use it. Tried quickly to create it outside of the functions, setting in 1, then trying to use it in 2, and the error message came up when I tried to check whether it didn't hold what was set in 1, meaning setting the variables like that won't work. So, the way it's set up is actually necessary.